src/chains/ethereum/transaction/src/eip1559-fee-market-transaction.ts

```diff
@@ -105,12 +105,6 @@ export class EIP1559FeeMarketTransaction extends RuntimeTransaction {
     } else {
       if (data.chainId) {
         this.chainId = Quantity.from(data.chainId);
-        if (this.common.chainId() !== this.chainId.toNumber()) {
-          throw new CodedError(
-            `Invalid chain id (${this.chainId.toNumber()}) for chain with id ${common.chainId()}.`,
-            JsonRpcErrorCode.INVALID_INPUT
-          );
-        }
       } else {
```

Comment:
Deleted because I need this to happen. Maaayyybbbe we could work around it, but I'm not sure it'll be a problem since `data.chainId` should only be coming from the database? :thinking:  hmmmm